### PR TITLE
Create the certificate with a subjectAltName

### PR DESCRIPTION
**Pull request by kroeckx**
Signed-off-by: Kurt Roeckx <kurt@roeckx.be>

### CODE REVIEW

**Comment by richvdh** [comment]
Please could you merge `develop`

**Comment by clokep** [comment]
Curious -- what's the reasoning for this?

**Comment by kroeckx** [comment]
> Curious -- what's the reasoning for this?

The subjectAltName is a mandatory field for such a certificate,
and there are implementations that rejects certificates without
it.

In this case, Conduit is using rust webpki when acting as client
to validate the certificate, and it rejects the certificate as
invalid.

File: lib/SyTest/SSL.pm [hunk 1]
```diff
@@ -59,11 +59,17 @@ sub create_ssl_cert
       "-subj", "/CN=$server_name",
    ) == 0 or die "openssl req failed $?";
 
+   # Create extenion file
```
Comment (richvdh) [inline lib/SyTest/SSL.pm:62]:
```suggestion
   # Create extension file
```

File: lib/SyTest/SSL.pm [hunk 2]
```diff
@@ -59,11 +59,17 @@ sub create_ssl_cert
...
+   my $ext_file = "$cert_file.ext";
+   open(my $fh, '>', $ext_file) or die "Could not open file '$filename' $!";
+   print $fh "subjectAltName=\@alts\n[alts]\nDNS=$server_name\n";
```
Comment (richvdh) [inline lib/SyTest/SSL.pm:65]: can we not just write this:

```suggestion
   print $fh "subjectAltName=DNS:$server_name\n";
```

**Review by richvdh — COMMENTED** [review]
seems sensible. question below

File: lib/SyTest/SSL.pm [hunk 3]
```diff
@@ -59,11 +59,17 @@ sub create_ssl_cert
       "-subj", "/CN=$server_name",
    ) == 0 or die "openssl req failed $?";
 
+   # Create extension file
+   my $ext_file = "$cert_file.ext";
+   open(my $fh, '>', $ext_file) or die "Could not open file '$filename' $!";
```
Comment (richvdh) [inline lib/SyTest/SSL.pm:64]:
```suggestion
   open(my $fh, '>', $ext_file) or die "Could not open file '$ext_file' $!";
```

**Review by richvdh — APPROVED** [review]
thanks!

**Pull request merged**